calyx-opt/src/passes/static_promotion.rs

```diff
@@ -226,12 +226,33 @@ impl StaticPromotion {
             })
             .collect();
 
-        let mut stop_loop = false;
-        given_opts.iter().for_each(|arg| {
-            if *arg == "stop_loop" {
-                stop_loop = true
+        // searching for "-x static-promotion:cycle-limit=200" and getting back "200"
+        let cycle_limit_str: Option<&str> = given_opts.iter().find_map(|arg| {
+            let split: Vec<&str> = arg.split('=').collect();
+            if let Some(str) = split.first() {
+                if str == &"cycle-limit" {
+                    return Some(split[1]);
+                }
             }
+            None
         });
+        // If they don't give a cycle_limit_str, then we set it to the Default.
+        // If they explicitly say "None" or "none", hten we set it to None.
+        // Makes things a bit confusing.
+        let cycle_limit = if cycle_limit_str.is_none() {
+            Some(33554432)
```

Comment:
How is this number chosen?

Reply:
Ah my bad should've explained that. 
I basically looked at what worked well for the LUT numbers for the polybench stuff I've been working on. 33554432=2^25 ended up working well as an upper bound.

Reply:
I think we need to be a little more principled about this perhaps. What happens for higher numbers? We want some argument for why a 25-bit counter being the maximum is a good default choice.

Reply:
Ok finally fixed this to default to none (i.e., no limit). We can always set `cycle-limit` though manually if we want.